### PR TITLE
Add the policy name to the audit logs tags when doing policy-based API calls. Add retention settings to tags

### PR DESCRIPTION
**Pull request by marktheunissen**
## Community Contribution License
All community contributions in this pull request are licensed to the project maintainers
under the terms of the [Apache 2 license](https://www.apache.org/licenses/LICENSE-2.0). 
By creating this pull request I represent that I have the right to license the 
contributions to the project maintainers under the Apache 2 license.

## Description

Based on customer request in subnet ticket 12167, two things:

- When creating audit logs for API calls that deal with policies, audit logs should include the policy name(s).
- When setting retention on a bucket, the days/years should be in the audit log too.

This PR creates audit JSON like this: 

SetPolicyForUserOrGroup 
`"tags":{"policyName":"consoleAdmin,readwrite"...`

When multiple policies are handled, it just presents them as a CSV list under the same tag "policyName".

And for retention API call:

`"retention":"Enabled: true, Mode: GOVERNANCE, Days: 1, Years: 0"}`

## Motivation and Context

customer request

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fminio%2fpull%2fNNNNN)

### CODE REVIEW

**Comment by shtripat** [comment]
Do we need to add this changes to `AttachDetachPolicyBuiltin` as well?

File: cmd/bucket-handlers.go [hunk 1]
```diff
@@ -1809,6 +1809,10 @@ func (api objectAPIHandlers) PutBucketObjectLockConfigHandler(w http.ResponseWri
 		return
 	}
 
+	// Audit log tags.
```
Comment (shtripat) [inline cmd/bucket-handlers.go:1812]: Which `mc` command covers this path?
`mc retention set governance` doesn't follow this rather it goes through `PutObjectRetention`

Comment (marktheunissen) [reply cmd/bucket-handlers.go:1812]: @shtripat Interesting.. I have been testing with `mc retention set --default governance 1d local/lockedbucket` and it is audit logging as follows:

`...."api":{"name":"PutBucketObjectLockConfig","bucket":"lockedbucket","status":"OK","statusCode":20...`

Comment (marktheunissen) [reply cmd/bucket-handlers.go:1812]: I will take a look at the PutObjectRetention handler too and see if I can trigger that

Comment (marktheunissen) [reply cmd/bucket-handlers.go:1812]: Ok I can add handling for PutObjectRetention, but just note that it's not specifically what the customer was asking for. Object retention tag will look like:

`"tags":{"retention":"Mode: GOVERNANCE, RetainUntilDate: 2024-12-15 01:24:26 +0000 UTC"`

**Comment by harshavardhana** [comment]
> Based on customer request in subnet ticket 12167, two things:

Subnet related PRs must go miniohq/eos first @marktheunissen as practice.

File: cmd/bucket-handlers.go (hunk 1 above)
Comment (shtripat) [reply cmd/bucket-handlers.go:1812]: > @shtripat Interesting.. I have been testing with `mc retention set --default governance 1d local/lockedbucket` and it is audit logging as follows:
> 
> `...."api":{"name":"PutBucketObjectLockConfig","bucket":"lockedbucket","status":"OK","statusCode":20...`

ok, let me try this

**Review by shtripat — APPROVED** [review]
lgtm, verified

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**